pkg/services/ngalert/api/api_ruler.go

```diff
@@ -141,10 +154,17 @@ func (srv RulerSrv) RouteDeleteAlertRules(c *contextmodel.ReqContext, namespaceU
 			return nil
 		}
 		// if none rules were deleted return an error.
+
 		// Check whether provisioned check failed first because if it is true, then all rules that the user can access (actually read via GET API) are provisioned.
 		if provisioned {
 			return errProvisionedResource
 		}
+
+		// If auth is false, then the user is not authorized to delete any of the rules.
+		if !auth {
+			return authz.NewAuthorizationErrorGeneric("delete any existing rules in the namespace")
```

Comment:
The only way this happens is from query permissions on groups. Can probably give a better hint as to what is going on in the error here.

Reply:
There's no good mechanism to show the missing permissions here, and considering this may be going away in the future I'm not sure if it's worth focusing on it too much. We could say something like "delete any existing rules in the namespace due to missing data source query permissions" though, WDYT?